### PR TITLE
feat: improve filter by users

### LINKED ISSUE #229: Improve user-search and filtering
We got new design for how to search for processes with users.
image

**Requirements**:
- [ ] filter processes on one or several users
- [ ] search for users to filter on
- [ ] the filter should use AND-logic. For example: selecting two users will display the processes where both those two users have access.

Back-end
- [x] new user search endpoint. (search on shortname untill #197 #148)
- [x] New user-filter-param for processes 
- [x] Remove `q-user-search-functionality`

Front-end
- [x] Implement new design and functionality (@Linedueo awaiting design) 

## TODO How to test

Fixes #226

### PR DESCRIPTION
**Pull request by SjoenH**
New design for filtering processes by users

closes #229

### CODE REVIEW

File: pages/processes/index.tsx [hunk 1]
```diff
@@ -1,36 +1,44 @@
...
   const router = useRouter();
-
+  const requiredUsers = router.query?.user
+    ?.toString()
+    .split(",")
+    .map((user) => parseInt(user));
+  const requiredLabels = router.query?.rl
+    ?.toString()
+    .split(",")
+    .map((label) => parseInt(label));
```
Comment (dansvend) [inline pages/processes/index.tsx:30]: Consider creating creating one/two helper function(s) and reuse this on the three pages

File: pages/processes/favourite.tsx [hunk 2]
```diff
@@ -1,32 +1,42 @@
...
   const router = useRouter();
-
+  const requiredUsers = router.query?.user
+    ?.toString()
+    .split(",")
+    .map((user) => parseInt(user));
+  const requiredLabels = router.query?.rl
+    ?.toString()
+    .split(",")
+    .map((label) => parseInt(label));
```
Comment (dansvend) [inline pages/processes/favourite.tsx:30]: Consider creating creating one/two helper function(s) and reuse this on the three pages

**Review by dansvend — APPROVED** [review]
Clean code👌 I added a couple of questions

File: components/FilterUserButton.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        color="primary"
+        variant="ghost"
+        style={{
+          fontSize: "14px",
+          fontWeight: "normal",
+          marginRight: "8px",
+        }}
+        onClick={() => setVisibleScrim(true)}
+      >
+        Filter by users{" "}
```
[the remark this answers is not in the document]
Comment (SjoenH) [reply components/FilterUserButton.tsx:25]: Good suggestion. Will fix this

File: components/Labels/FilterLabelButton.tsx [hunk 4]
```diff
@@ -1,9 +1,15 @@
 import { Button, Scrim } from "@equinor/eds-core-react";
 import React, { useState } from "react";
+
```
[the remark this answers is not in the document]
Comment (SjoenH) [reply components/Labels/FilterLabelButton.tsx:3]: I'm automatically sorting my imports. And I guess it likes to separate them. Will take a look.

File: components/Labels/FilterLabelButton.tsx [hunk 5]
```diff
@@ -16,7 +22,8 @@ export default function FilterLabelButton(): JSX.Element {
         }}
         onClick={() => setVisibleScrim(true)}
       >
-        Filter by label
+        Filter by label{" "}
```
[the remark this answers is not in the document]
Comment (SjoenH) [reply components/Labels/FilterLabelButton.tsx:25]: Will do

File: components/Labels/FilterLabelButton.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (dansvend) [reply components/Labels/FilterLabelButton.tsx:3]: If that's standard functionality for a code formatter I'm all for having it like you have it

File: components/FilterUserButton.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (SjoenH) [reply components/FilterUserButton.tsx]: Fixed

File: pages/processes/favourite.tsx [hunk 6]
```diff
@@ -1,32 +1,35 @@
...
   const query = useQuery(
     ["favProjects", page, "isFavourite", ...Object.values(router.query)],
     () =>
       getProjects({
         page,
         items: itemsPerPage,
         onlyFavorites: true,
-        ...router.query,
+        ru: stringToArray(router.query.user),
+        rl: stringToArray(router.query.rl),
```
Comment (SjoenH) [inline pages/processes/favourite.tsx:32]: I believe this is cleaner. Let me know what you think :) @dansvend

Comment (dansvend) [reply pages/processes/favourite.tsx:32]: Perfect👌

**Review by dansvend — APPROVED** [review]
Clean👌

**Comment by SjoenH** [comment]
Changed the text on no results to: “No processes match your search criteria.”
Added the active-filter-section to my-page and favourites-page.
Fixed crash when you use your own user as a filter while on my-page.
Fixed the search-field.
Not sure why the font is misbehaving… I'll fix it later when upgrading some packages
Upgrade everything #354

**Review by dansvend — APPROVED** [review]
Looks good💪

**Pull request merged**